new blobber aggregates

devyetii opened the pull request:
## Fixes

## Changes
- Add `total_storage_income`, `total_read_income` and `total_slashed_stake` to blobbers, blobber_snapshots and blobber_aggregates
## Need to be mentioned in CHANGELOG.md?

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

devyetii committed:
new blobber aggregates

dabasov reviewed (commented):
looks good, add other missing aggregates

devyetii committed:
authorizer total_mint and total_burn

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
unit test for new blobber fields

devyetii committed:
unit test to authorizer mint/burn

devyetii committed:
complete unit tests

fix blobberSpecificRevenue

fix unit test

fix unit test

fix unit test

fix unit test

fix unit test

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
Complete aggregate/snapshot testing based on the changes

devyetii committed:
fix migration numbering

devyetii committed:
fix stakepool

devyetii committed:
fix unit test

devyetii committed:
fix unit tests

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

service-0chain commented:
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | 2188-new-aggregates  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

dabasov committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
Merge branch '2188-new-aggregates' of github.com:0chain/0chain into 2188-new-aggregates

dabasov commented on code/go/0chain.net/chaincore/chain/state/state_context.go:
```diff
@@ -268,7 +269,28 @@ func (sc *StateContext) AddMint(m *state.Mint) error {
+}
+
+// AddBridgeMint - add bridge mint for the authorizers
+func (sc *StateContext) AddBridgeMint(m *state.Mint) error {
```
do not extend contex with new method, you can filter bridge burns by sender or emit additional message

dabasov replied:
new event here https://github.com/0chain/0chain/pull/2230

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
Remove state context new function

devyetii committed:
Merge branch '2188-new-aggregates' of github.com:0chain/0chain into 2188-new-aggregates

devyetii committed:
fix unit tests

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
fix unit test

devyetii committed:
Merge branch '2188-new-aggregates' of github.com:0chain/0chain into 2188-new-aggregates

dabasov committed:
Merge branch 'staging' into 2188-new-aggregates

dabasov commented on code/go/0chain.net/smartcontract/zcnsc/mint.go:
```diff
@@ -159,6 +159,20 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
+			Minter:     gn.ID,
+			ToClientID: sig.ID,
+			Amount:     share,
+			IsBridge:   true,
```
do not extend mint as well, since Mint is something common, and it shouldn't know anything about smart contract logic like bridging. you can emit burn event right from here, check https://github.com/0chain/0chain/pull/2230 here is this event already created

devyetii replied:
I will do

devyetii replied:
Re 2230, If it's merged before this PR, I'll remove the definition of the event from here.

devyetii committed:
dispatch event directly

devyetii committed:
rename the events

devyetii committed:
fix unit test

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
use single event for minting

devyetii committed:
fix unit test

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
fix unit test

add log

add log

log

calculate rank

Bump gorm.io/driver/sqlite from 1.4.4 to 1.5.0 in /code/go/0chain.net

Bumps [gorm.io/driver/sqlite](https://github.com/go-gorm/sqlite) from 1.4.4 to 1.5.0.
- [Release notes](https://github.com/go-gorm/sqlite/releases)
- [Commits](https://github.com/go-gorm/sqlite/compare/v1.4.4...v1.5.0)

---
updated-dependencies:
- dependency-name: gorm.io/driver/sqlite
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

log

clean logs

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

dabasov committed:
Merge branch 'staging' into 2188-new-aggregates

dabasov committed:
Merge branch 'staging' into 2188-new-aggregates

Manali-Jain-squareops committed:
golangci-lint fix

dabasov commented on code/go/0chain.net/smartcontract/dbs/event/stakepool.go:
```diff
@@ -138,6 +138,10 @@ func (edb *EventDb) rewardUpdate(spus []dbs.StakePoolReward, round int64) error
 			return fmt.Errorf("could not rewards providers: %v", err)
 		}
+
+		if err := edb.blobberSpecificRevenue(spus); err != nil {
```
move this code a level higher, we should not keep any provider specific code here, since stakepool.go contains only common code, in process.go you can call different handler for this blobber reward

dabasov commented on code/go/0chain.net/smartcontract/dbs/event/authorizer.go:
```diff
@@ -187,3 +229,7 @@ func mergeUpdateAuthorizerTotalUnStakesEvents() *eventsMergerImpl[Authorizer] {
 	return newEventsMerger[dbs.DbHealthCheck](TagAuthorizerHealthCheck, withUniqueEventOverwrite())
 }
+
+func mergeAuthorizerBurnEvents() *eventsMergerImpl[state.Burn] {
```
move merger for mints here

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
resolve comments

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

dabasov committed:
Merge branch 'staging' into 2188-new-aggregates

dabasov reviewed (approved):
lgtm

devyetii committed:
Merge branch 'staging' into 2188-new-aggregates

devyetii committed:
fix unit test

dabasov reviewed (approved):
lgtm
